float32 roll
float32 pitch
float32 grasp